.circleci/config.yml

```diff
@@ -53,32 +42,19 @@ jobs:
       - checkout
       - run:
           name: Prepare extra directories
-          command: |
-            sudo mkdir /floodgate
-            sudo chmod 777 /floodgate
-            mkdir /floodgate/bin
+          command: sudo .cilibs/prepare_extra_directories.sh 
       - run:
           name: Setup swagger-codegen
-          command: |
-            SWAGGER_VERSION=$(cat gateapi/.swagger-codegen/VERSION)
-            wget https://repo1.maven.org/maven2/io/swagger/swagger-codegen-cli/${SWAGGER_VERSION}/swagger-codegen-cli-${SWAGGER_VERSION}.jar -O swagger-codegen-cli.jar
-            wget https://repo1.maven.org/maven2/io/swagger/swagger-codegen-cli/${SWAGGER_VERSION}/swagger-codegen-cli-${SWAGGER_VERSION}.jar.sha1 -O swagger-codegen-cli.jar.sha1
-            echo ' swagger-codegen-cli.jar' >> swagger-codegen-cli.jar.sha1
-            sha1sum -c swagger-codegen-cli.jar.sha1 
-            mv swagger-codegen-cli.jar /floodgate/bin/
+          command: sudo .cilibs/setup_swagger_codegen.sh 
       - run:
           name: Get gate code
-          command: git clone https://github.com/spinnaker/gate.git -b << parameters.gate_api_branch >> /floodgate/gate
+          command: .cilibs/get_gate_code.sh << parameters.gate_api_branch >>
       - run:
           name: Generate swagger.json
-          command: |
-            cd /floodgate/gate
-            ./gradlew clean
-            ./gradlew gate-web:test --tests *GenerateSwagger* --max-workers 2
-            cat gate-web/swagger.json | json_pp > ./gate-swagger.json
+          command: .cilibs/generate_swagger_json.sh
       - run:
           name: Generate gateapi go code
-          command: java -jar /floodgate/bin/swagger-codegen-cli.jar generate -l go -i /floodgate/gate/gate-swagger.json -o /floodgate/gateapi
+          command: .cilibs/generate_gateapi_go_code.sh
```

Comment:
Maybe also this can be merged with swagger part :)

Reply:
Done